pkg/controllers/workapplier/availability_tracker.go

```diff
@@ -247,6 +252,38 @@ func trackPDBAvailability(curObj *unstructured.Unstructured) (ManifestProcessing
 	return ManifestProcessingAvailabilityResultTypeNotYetAvailable, nil
 }
 
+// trackServiceExportAvailability tracks the availability of a service export in the member cluster.
+// It is available if the ServiceExportValid condition is true (will be false if annotation value is invalid).
+// If the weight is not 0, ServiceExportValid condition must be true and the ServiceExportConflict condition must be false.
+func trackServiceExportAvailability(curObj *unstructured.Unstructured) (ManifestProcessingAvailabilityResultType, error) {
+	var svcExport fleetnetworkingv1alpha1.ServiceExport
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(curObj.Object, &svcExport); err != nil {
+		return ManifestProcessingAvailabilityResultTypeFailed, controller.NewUnexpectedBehaviorError(err)
+	}
+
+	// Check if ServiceExport is valid and up to date
+	validCond := meta.FindStatusCondition(svcExport.Status.Conditions, string(fleetnetworkingv1alpha1.ServiceExportValid))
+	if validCond == nil || svcExport.Generation != validCond.ObservedGeneration ||
+		meta.IsStatusConditionFalse(svcExport.Status.Conditions, string(fleetnetworkingv1alpha1.ServiceExportValid)) {
+		klog.V(2).InfoS("Still need to wait for ServiceExport to be available", "serviceExport", klog.KObj(curObj))
+		return ManifestProcessingAvailabilityResultTypeNotYetAvailable, nil
+	}
+
+	weight := svcExport.Annotations[objectmeta.ServiceExportAnnotationWeight]
```

Comment:
i just realize that, we need to validate the annotation here too. Cause updating the annotation won't change the generation. We cannot reply on the valid condition when cx only updates the weight.

I will expose this util https://github.com/Azure/fleet-networking/blob/main/pkg/controllers/member/serviceexport/controller.go#L405, so you can use here.

Reply:
please use this util to validate the annotation here. https://github.com/Azure/fleet-networking/pull/271/files